### PR TITLE
Safety moth

### PR DESCRIPTION
**Pull request by Horatio22**
## About The Pull Request

This adds safety moth ERT

## Why It's Good For The Game

This pays homage to Saftey Moth, which originated on Fulp. It also gives us an ERT that we can send in to fix engineering when the SM inevitably gets wrecked by people throwing themselves in at round start. It spawns with pacifism and without the usual energy gun, so this will (hopefully) avoid some of the valid hunting issues that ERTs often have.

## Changelog
:cl:
add: Safety Moth ERT
/:cl:

### CODE REVIEW

**Comment by Horatio22** [comment]
Thank you @GuillaumePrata for putting up with my smooth monkey brain and walking me through my first PR :)

File: fulp_modules/safety_moth/safety_moth.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/datum/ert/safety_moth
+    mobtype = /mob/living/carbon/human/species/moth
+    leader_role = /datum/antagonist/ert/safety_moth
+    enforce_human = FALSE
+    roles = /datum/antagonist/ert/safety_moth
+    mission = "ensure that proper safety protocols are being followed by the crew."
```
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:6]:
```suggestion
    mission = "Ensure that proper safety protocols are being followed by the crew."

```

File: fulp_modules/safety_moth/safety_moth.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/datum/antagonist/ert/safety_moth/on_gain()
+    forge_objectives()
+    . = ..()
+    equip_official()
+    ADD_TRAIT(owner.current, TRAIT_PACIFISM, JOB_TRAIT)
+
+/datum/antagonist/ert/safety_moth/greet()
+
+    to_chat(owner, "<B><font size=3 color=green>You are the [name].</font></B>")
+    to_chat(owner, "You are being sent on a mission to [station_name()] by Nanotrasen's Operational Safety Department. Ensure the crew is following all proper safety protocols. Board the shuttle when your team is ready.")
```
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:29]: "your team"? isn't this just one moth?

File: fulp_modules/safety_moth/safety_moth.dm [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    opendoors = FALSE
+
+
+/datum/antagonist/ert/safety_moth
+    name = "Safety Moth"
+    outfit = /datum/outfit/centcom/ert/engineer/safety_moth
+    role = "Safety Moth"
+    show_in_antagpanel = TRUE
+    show_to_ghosts = TRUE
+    antag_moodlet = /datum/mood_event/focused
```
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:18]:
```suggestion
```

File: fulp_modules/safety_moth/safety_moth.dm [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    to_chat(owner, "<B><font size=3 color=green>You are the [name].</font></B>")
+    to_chat(owner, "You are being sent on a mission to [station_name()] by Nanotrasen's Operational Safety Department. Ensure the crew is following all proper safety protocols. Board the shuttle when your team is ready.")
+
+
+/datum/antagonist/ert/safety_moth/proc/equip_official()
+    var/mob/living/carbon/human/H = owner.current
+    if(!istype(H))
+        return
+    H.equipOutfit(/datum/outfit/centcom/centcom_official)
+    H.set_species(/datum/species/moth)
```
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:37]:
```suggestion
```
appears to be redundant, and this tries to equip the wrong outfit anyway

File: fulp_modules/safety_moth/safety_moth.dm [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+/datum/antagonist/ert/safety_moth/create_team(datum/team/new_team)
+    if(istype(new_team))
+        ert_team = new_team
```
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:41]:
```suggestion
```
this is already inherited from /datum/antagonist/ert

File: fulp_modules/safety_moth/safety_moth.dm (hunk 3 above)
Comment (ATH1909) [inline fulp_modules/safety_moth/safety_moth.dm:18]:
```suggestion
```
this is already inherited from /datum/antagonist/ert

**Comment by ATH1909** [comment]
>  It spawns with pacifism and without the usual energy gun

~~Aw sweet, a defenceless loot pinata! Time to loot some gamer armor!~~

I forget sometimes that you have a very restrictive antag policy.

File: fulp_modules/safety_moth/safety_moth.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    to_chat(owner, "<B><font size=3 color=green>You are the [name].</font></B>")
+    to_chat(owner, "You are being sent on a mission to [station_name()] by Nanotrasen's Operational Safety Department. Ensure the crew is following all proper safety protocols. Board the shuttle when your team is ready.")
+
+
+/datum/antagonist/ert/safety_moth/proc/equip_official()
+    var/mob/living/carbon/human/H = owner.current
+    if(!istype(H))
+        return
+    H.set_species(/datum/species/moth)
+    H.equipOutfit(outfit)
```
Comment (JohnFulpWillard) [inline fulp_modules/safety_moth/safety_moth.dm:34]: Remove this line, they inherit the equipOutfit through /datum/antagonist/ert

File: fulp_modules/safety_moth/safety_moth.dm [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+
+/datum/antagonist/ert/safety_moth
+    name = "Safety Moth"
+    outfit = /datum/outfit/centcom/ert/engineer/safety_moth
+    role = "Safety Moth"
+
+/datum/antagonist/ert/safety_moth/on_gain()
+    forge_objectives()
+    . = ..()
```
Comment (JohnFulpWillard) [inline fulp_modules/safety_moth/safety_moth.dm:19]: This should be at the end I think, someone reported safety moths not being pacifists, that might be why

**Review by JohnFulpWillard — APPROVED** [review]

**Comment by JohnFulpWillard** [comment]
Also, I said in the discord, but replace the spaces with tabs

**Comment by JohnFulpWillard** [comment]
> 
> 
> > It spawns with pacifism and without the usual energy gun
> 
> ~Aw sweet, a defenceless loot pinata! Time to loot some gamer armor!~
> 
> I forget sometimes that you have a very restrictive antag policy.

time to give them close quarter pacification

File: fulp_modules/safety_moth/safety_moth.dm [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    equip_official()
+    ADD_TRAIT(owner.current, TRAIT_PACIFISM, JOB_TRAIT)
```
Comment (JohnFulpWillard) [inline fulp_modules/safety_moth/safety_moth.dm:21]: shouldnt this be 'ADD_TRAIT(owner, TRAIT_PACIFISM, JOB_TRAIT)'?

**Review by Horatio22 — COMMENTED** [review]
> Also, I said in the discord, but replace the spaces with tabs

Should be fixed (I think)

File: fulp_modules/safety_moth/safety_moth.dm (hunk 6 above)
Comment (Horatio22) [reply fulp_modules/safety_moth/safety_moth.dm]: Worked! Thanks!

File: fulp_modules/safety_moth/safety_moth/safety_moth.dm [hunk 9]
```diff
@@ -0,0 +1,99 @@
+/datum/ert/safety_moth
```
Comment (JohnFulpWillard) [inline fulp_modules/safety_moth/safety_moth/safety_moth.dm:1]: This file isnt in the tgstation.dme, and why is /datum/ert/safety_moth in both files?

**Comment by FernandoJ8** [comment]
> time to give them close quarter pacification

Close Quarters Calming

**Pull request closed without merging**